### ISSUE TITLE
Add `x-elastic-product-origin` header to all places that make requests to Elasticsearch

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-output-elasticsearch#1191]
The Logstash Elasticsearch output plugin makes HTTP requests to the following Elasticsearch endpoints:

```
GET /
HEAD /
GET /_license
POST _bulk?filter_path=..
HEAD /_index_template/<index_name>
PUT /_index_template/<index_name>
HEAD /<index_name>
PUT /<index_name>
GET /_ilm/policy/logstash-policy
PUT /_ilm/policy/logstash-policy
GET /_nodes/http (sniffing)
```

Notes:

```parent``` setting no longer works (probably hasn't for a long time)

Each of these calls should include the x-elastic-product-origin header to the HTTP request made to Elasticsearch

**Comment by robbavey** [comment]
Updated to correct the header name to x-elastic-product-origin

**Comment by donoghuc** [comment]
Publish job https://github.com/logstash-plugins/publisher/actions/runs/12241803704
Shipped in [11.22.10](https://rubygems.org/gems/logstash-output-elasticsearch/versions/11.22.10-java)
